File: app/src/main/java/org/pyload/android/client/Preferences.java
package org.pyload.android.client;

import android.os.Bundle;
import android.view.MenuItem;

import org.pyload.android.client.module.AppCompatPreferenceActivity;

import androidx.appcompat.app.ActionBar;

public class Preferences extends AppCompatPreferenceActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.preferences);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setHomeButtonEnabled(true);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
        }
        return true;
    }
}

File: app/src/main/java/org/pyload/android/client/Preferences.kt
package org.pyload.android.client

import android.os.Bundle
import android.view.MenuItem
import org.pyload.android.client.module.AppCompatPreferenceActivity
import org.pyload.android.client.utils.NavigationUtil

class Preferences : AppCompatPreferenceActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.preferences)

        supportActionBar?.run {
            setHomeButtonEnabled(true)
            setDisplayHomeAsUpEnabled(true)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> NavigationUtil.navigateUp(this)
        }
        return true
    }
}